internal/services/compute/shared_image_version_resource.go

```diff
@@ -511,6 +574,169 @@ func expandSharedImageVersionTargetRegions(d *pluginsdk.ResourceData) (*[]galler
+				Type:     pluginsdk.TypeString,
+				Required: true,
+			},
+			"key_type": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice(galleryimageversions.PossibleValuesForUefiKeyType(), false),
+			},
+		},
+	}
+}
+
+func expandUefiSettings(d *pluginsdk.ResourceData) *galleryimageversions.GalleryImageVersionUefiSettings {
+	if uefiSettings, ok := d.GetOk("uefi_settings"); ok {
+		settings := uefiSettings.([]interface{})
+		if len(settings) == 0 {
+			return nil
+		}
+
+		us := settings[0].(map[string]interface{})
+		return &galleryimageversions.GalleryImageVersionUefiSettings{
+			SignatureTemplateNames: expandStringList(us["signature_template_names"].([]interface{})),
+			AdditionalSignatures:   expandAdditionalSignatures(us["additional_signatures"]),
+		}
+	}
+	return nil
+}
+
+func expandAdditionalSignatures(input interface{}) *galleryimageversions.UefiKeySignatures {
+	if input == nil {
+		return nil
+	}
+
+	data := input.([]interface{})[0].(map[string]interface{})
+	return &galleryimageversions.UefiKeySignatures{
+		Db:  expandUefiKeyList(data["db"]),
+		Dbx: expandUefiKeyList(data["dbx"]),
+		Kek: expandUefiKeyList(data["kek"]),
+		Pk:  expandSingleUefiKey(data["pk"]),
+	}
+}
+
+func expandUefiKeyList(input interface{}) *[]galleryimageversions.UefiKey {
+	if input == nil {
+		return nil
+	}
+
+	keys := input.([]interface{})
+	result := make([]galleryimageversions.UefiKey, len(keys))
+	for i, v := range keys {
+		result[i] = expandUefiKey(v.(map[string]interface{})) // Correctly returns a value, not a pointer
+	}
+	return &result
+}
+
+func expandSingleUefiKey(input interface{}) *galleryimageversions.UefiKey {
+	if input == nil {
+		return nil
+	}
+
+	keys := input.([]interface{})
+	if len(keys) == 0 {
+		return nil
+	}
+
+	data := keys[0].(map[string]interface{})
+	return &galleryimageversions.UefiKey{
+		Type:  pointer.To(galleryimageversions.UefiKeyType(data["key_type"].(string))),
+		Value: &[]string{data["certificate_data"].(string)},
+	}
+}
+
+func expandUefiKey(data map[string]interface{}) galleryimageversions.UefiKey {
+	return galleryimageversions.UefiKey{
+		Type:  pointer.To(galleryimageversions.UefiKeyType(data["key_type"].(string))),
+		Value: &[]string{data["certificate_data"].(string)},
+	}
+}
+
+func expandStringList(input []interface{}) *[]galleryimageversions.UefiSignatureTemplateName {
+	result := make([]galleryimageversions.UefiSignatureTemplateName, len(input))
+	for i, v := range input {
+		result[i] = galleryimageversions.UefiSignatureTemplateName(v.(string))
+	}
+	return &result
+}
+
+func flattenUefiSettings(input *galleryimageversions.GalleryImageVersionUefiSettings) []interface{} {
+	if input == nil {
+		return []interface{}{}
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"signature_template_names": *input.SignatureTemplateNames,
+			"additional_signatures": []interface{}{
+				flattenAdditionalSignatures(input.AdditionalSignatures),
+			},
```

Comment:
The return type of `flattenAdditionalSignatures` should be `[]interface{}`
```suggestion
			"additional_signatures": flattenAdditionalSignatures(input.AdditionalSignatures),
```

Reply:
:+1: